### DISCUSSION TITLE
Add ability to tag photos

### DISCUSSION THREAD

**Discussion by SaintsandSailors** [ente-io/ente#543] [Enhancements]

**Comment by ph00lt0** [comment]
What's the difference from albums/favourites?

**Comment by SaintsandSailors** [comment]
> What's the difference from albums/favourites?

Fair question. I suppose the two could function equivalently, but in my mind, albums and tags are not synonymous. I'd like to be able to tag photos with, for example, "golf", but I don't want to create an album for golf. If I did create an album for golf and all of the other tags, I'd have more albums than I'd like. 

Additionally, without facial recognition, I'd like to be able to tag photos with those who are in the photos, but again, I don't want to dedicate an album for photos with (for example) "mom" in them. 

Does that make sense?

**Reply by leikarnes** [reply]
I second this. Also a photo can have many tags

**Comment by chuegel** [comment]
Would also love to see tag support implemented.

**Comment by thz** [comment]
I would extend this to albums as well. Photos and albums could have multiple tags.
Tags should be a filter/search criterion.
This might lead to a situation where tags could possibly replace albums similarly to how gmail uses tags to implement folders.

**Comment by githubkusi** [comment]
Duplicates
* #3596
* #4045
* #5082
* #955 
* #543

**Comment by mrclac14** [comment]
That would be really great, ideally supported via XMP. This, as I understand, is what is being used under the hood in iOS, and I, like many others, am also using it in my own workflow with different tools. I just like the granularity and no-frills filter options that tags are giving me.

**Comment by Blurazzle** [comment]
I’m not well‑versed in photo‑editing software, but it seems to me that XMP is the default format for metadata. It is also used by Immich. Supporting XMP would improve interoperability with Immich and other photo‑editing software.

**Comment by bverkron** [comment]
Adding my vote for reading existing metadata. It's surprising to me that photo sharing platforms like Ente don't support this early on. Immich does but Nextcloud does not and neither does Ente apparently. It's too bad as this is a show stopper for me and my sharing workflow. I also have a heavy amount of time spent in keywording people and other metadata in Lightroom I want to users that I share with to be able to search on.

**Comment by zilexa** [comment]
Now it finally makes sense Ente has no way of organising albums. Its because there is no tagging feature at all. Not for photos. Not for albums. 🤦🏽‍♂️ Really wish I hadn't paid for 1TB and migrated 260 albums/316GB already..

**Comment by r-cheologist** [comment]
Adding a pointer here too — [#1910](https://github.com/ente-io/ente/discussions/1910) has a writeup on why XMP import matters specifically for migrating from tools like digiKam/Lightroom (face regions via the MWG schema, plus keyword tags via `dc:subject`/`lr:hierarchicalSubject`, which seems directly relevant to how tags get populated here).
